### PR TITLE
feat: Espresso design system + Storybook

### PR DESCRIPTION
**Pull request by netchampfaris**
image

### Button.vue

`appearance` prop removed in favor of more flexible props for appearance: `variant` and `theme`

| Before | After |
|---|---|
|`<Button appearance="primary">`|`<Button variant="solid">`|
|`<Button appearance="danger">`|`<Button variant="solid" theme="red">`|
|`<Button appearance="minimal">`|`<Button variant="ghost">`|
|`<Button appearance="white">`|`<Button variant="outline">`|
|`<Button icon-left="plus">`|`<Button><template #prefix><FeatherIcon name="plus" class="w-4" /></template></Button`|
|`<Button icon-right="plus">`|`<Button><template #suffix><FeatherIcon name="plus" class="w-4" /></template></Button`|

### Dialog.vue
| Before | After |
|---|---|
|`<Dialog :options="{ actions: [label: 'Save', handler: fn ] }">`|`<Dialog :options="{ actions: [label: 'Save', onClick: fn ] }">`|

### Dropdown
| Before | After |
|---|---|
|`<Dropdown :items="[label: 'Save', handler: fn ]">`|`<Dropdown :items="[label: 'Save', onClick: fn ]">`|

### vite config

```diff
import { defineConfig } from 'vite'
import vue from '@vitejs/plugin-vue'
+ import frappeui from 'frappe-ui/vite'
- import { getProxyOptions } from 'frappe-ui/src/utils/vite-dev-server'

export default defineConfig({
  plugins: [
+  frappeui(),
    vue(),
    ...
  ],
- server: {
-   port: 8080,
-   proxy: getProxyOptions({ port: webserver_port }),
- },
  build: {
    ...
  },
})

```

### CODE REVIEW

**Comment by netchampfaris** [comment]
## Breaking changes

### Button.vue

`appearance` prop removed in favor of more flexible props for appearance: `variant` and `theme`

| Before | After |
|---|---|
|`<Button appearance="primary">`|`<Button variant="solid">`|
|`<Button appearance="danger">`|`<Button variant="solid" theme="red">`|
|`<Button appearance="minimal">`|`<Button variant="ghost">`|
|`<Button appearance="white">`|`<Button variant="outline">`|

### Dialog.vue
| Before | After |
|---|---|
|`<Dialog :options="{ actions: [label: 'Save', handler: fn ] }">`|`<Dialog :options="{ actions: [label: 'Save', onClick: fn ] }">`|

**Comment by NagariaHussain** [comment]
Deploy preview is broken btw:

CleanShot 2023-08-08 at 16 51 31@2x

File: src/utils/fileUploadHandler.ts [hunk 1]
```diff
@@ -87,8 +88,11 @@ class FileUploadHandler {
           reject(error);
         }
       };
-      xhr.open('POST', '/api/method/upload_file', true);
+
+      const uploadEndpoint = options.upload_endpoint || '/api/method/upload_file';
```
Comment (netchampfaris) [inline src/utils/fileUploadHandler.ts:92]: @NagariaHussain Just `url` instead of `upload_endpoint`?

File: src/utils/frappeRequest.js [hunk 2]
```diff
@@ -90,14 +91,6 @@ export function frappeRequest(options) {
             : ['Internal Server Error']
         }
 
-        if (options.onError) {
```
Comment (nextchamp-saqib) [inline src/utils/frappeRequest.js:93]: @netchampfaris 
This was good for catching all the errors at the request level. 
For eg. creating a toast if any request failed or routing to `/login` if unauthorized

**Pull request merged**